mlem/contrib/kubernetes/base.py

```diff
@@ -0,0 +1,126 @@
+import os
+from typing import ClassVar, Optional
+
+from kubernetes import config
+
+from mlem.config import project_config
+from mlem.core.objects import (
+    DeployState,
+    DeployStatus,
+    MlemDeployment,
+    MlemEnv,
+    MlemModel,
+)
+from mlem.runtime.server import Server
+from mlem.ui import EMOJI_BUILD, EMOJI_OK, echo, set_offset
+
+from ..docker.base import (
+    DockerEnv,
+    DockerImage,
+    DockerImageBuilder,
+    DockerRegistry,
+    generate_docker_container_name,
+)
+from .context import K8sYamlBuildArgs, K8sYamlGenerator
+
+POD_STATE_MAPPING = {
+    "Pending": DeployStatus.STARTING,
+    "Running": DeployStatus.RUNNING,
+    "Succeeded": DeployStatus.STOPPED,
+    "Failed": DeployStatus.CRASHED,
+    "Unknown": DeployStatus.UNKNOWN,
+}
+
+
+class K8sDeploymentState(DeployState):
+    type: ClassVar = "kubernetes"
+    image: Optional[DockerImage]
+    pod_name: Optional[str]
+    kube_config_file_path: Optional[str] = None
+
+    def get_client(self):
+        config.load_kube_config(
+            config_file=self.kube_config_file_path
+            or os.getenv("KUBECONFIG", default="~/.kube/config")
+        )
+
+
+class K8sDeployment(MlemDeployment):
+    type: ClassVar = "kubernetes"
+    server: Optional[Server] = None
```

Comment:
Better not to mix classvars with fields visually

Reply:
hmm, this is done everywhere -- even for classes like `DockerContainer`, `DockerDirBuilder`, `DockerImageBuilder`, etc.

Reply:
then we should move them in those classes too

Reply:
pls fix

Reply:
also, why server is optional? can you deploy without it?

Reply:
separated visually is done, I guess one can deploy it because of 
```
meta.server or project_config(meta.loc.project if meta.is_saved else None).server
```

where this server (from the 2nd part) comes from
```
@property
    def server(self):
        from mlem.runtime.server import Server

        if not self.SERVER:
            return parse_obj_as(Server, {"type": "fastapi"})
        return parse_obj_as(Server, self.SERVER)
```

so it will work even if `meta.server` is None